codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -174,25 +175,26 @@ public void readFrame(ChannelHandlerContext ctx, ByteBuf input, Http2FrameListen
         }
     }
 
-    private void processHeaderState(ByteBuf in) throws Http2Exception {
+    private boolean preProcessFrame(ByteBuf in) throws Http2Exception {
+        // Start pre-processing the frame by reading the necessary data
+        // in common between all frame types
         if (in.readableBytes() < FRAME_HEADER_LENGTH) {
-            // Wait until the entire frame header has been read.
-            return;
+            // Wait until the entire framing section has been read.
+            return false;
         }
-
-        // Read the header and prepare the unmarshaller to read the frame.
         payloadLength = in.readUnsignedMedium();
+        frameType = in.readByte();
```

Comment:
Consider adding a comment for future devs why we read the full header before throwing an excepting on case `payloadLength > maxFrameSize`

Reply:
I'm looking at this again and since it's a connectionError which closes the connection I don't think we have to read the full headers before throwing the exception so I'm undoing these changes